core/jobs/batch_jobs/manual_voice_artist_names_job.py

```diff
@@ -0,0 +1,458 @@
+
+                old_recorded_voiceovers = change_dict.get('old_value')
+                old_voiceovers_mapping = old_recorded_voiceovers.get(
+                    'voiceovers_mapping')
+
+                # Getting the difference between old and new commit changes.
+                voiceovers_mapping = (
+                    self.get_voiceover_from_recorded_voiceover_diff(
+                        new_voiceovers_mapping, old_voiceovers_mapping))
+
+                for content_id, lang_code_to_voiceover_dict in (
+                    voiceovers_mapping.items()):
+
+                    # If some old commit models contain voiceovers for
+                    # contents that are not part of the current version,
+                    # then we should skip the content.
+                    if content_id not in exploration_mapping:
+                        continue
+
+                    for lang_code, voiceover_dict in (
+                        lang_code_to_voiceover_dict.items()):
+
+                        # Updating voice artist metadata information for the
+                        # given voice artist.
+                        self.update_voice_artist_data(
+                            voice_artist_id,
+                            exploration_id,
+                            lang_code,
+                            voiceover_dict
+                        )
+
+                        # If some old commit models contain voiceovers for
+                        # contents in some languages that are not part of the
+                        # current version, then we should skip the language
+                        # code.
+                        if lang_code not in exploration_mapping.get(content_id):
+                            continue
+
+                        # If the exploration mapping already includes the voice
+                        # artist ID for the specified language code, we should
+                        # skip it, as we are iterating through commit log models
+                        # in reverse order, meaning that in earlier versions,
+                        # the voice artist ID would have already been added.
+                        if exploration_mapping[content_id][lang_code] != '':
+                            continue
+
+                        exploration_mapping[content_id][lang_code] = (
+                            voice_artist_id)
+
+    def get_voice_artist_metadata_models(self):
+        """The method creates a list of voice artist metadata models.
+
+        Returns:
+            list(VoiceArtistMetadataModel). A list of voice artist
+            metadata models.
+        """
+        voice_artist_data_models = []
+
+        with datastore_services.get_ndb_context():
+            for voice_artist_id, voiceover_mapping in (
+                    self.voice_artist_id_to_voiceovers_data.items()):
+
+                language_code_to_voiceovers = (
+                    voiceover_mapping['language_code_to_voiceovers'])
```

Comment:
In general when you say "voiceovers" here do you mean a voiceover domain object or the URL to the voiceover? It currently sounds like the former to me.

Reply:
Removed this part.